src/dataflow/src/render/debezium.rs

```diff
@@ -118,6 +124,273 @@ where
+                                Err(err) => {
+                                    output.session(&binary_cap).give((
+                                        Err(err.into()),
+                                        *binary_cap.time(),
+                                        1,
+                                    ));
+                                    let _ = data_buffer.pop_front();
+                                    continue;
+                                }
+                            };
+                            let value = match result.value {
+                                Some(Ok(value)) => value,
+                                Some(Err(err)) => {
+                                    output.session(&binary_cap).give((
+                                        Err(err.into()),
+                                        *binary_cap.time(),
+                                        1,
+                                    ));
+                                    let _ = data_buffer.pop_front();
+                                    continue;
+                                }
+                                None => {
+                                    let _ = data_buffer.pop_front();
+                                    continue;
+                                }
+                            };
+
+                            let tx_id = value.iter().nth(transaction_idx).unwrap().unwrap_list().iter().nth(tx_id_idx).unwrap().unwrap_str();
+
+                            let tx_time: Timestamp = match tx_mapping.get(tx_id) {
+                                Some(time) => *time,
+                                None => break,
+                            };
+
+                            // We're committed to processing it if we have a tx_time
+                            let _ = data_buffer.pop_front();
+
+                            let partition_dedup = dedup_state
+                                .entry(result.partition.clone())
+                                .or_insert_with(|| {
+                                    DebeziumDeduplicationState::new(envelope.clone())
+                                });
+                            let should_use = match partition_dedup {
+                                Some(ref mut s) => {
+                                    let res = s.should_use_record(
+                                        key,
+                                        &value,
+                                        result.position,
+                                        result.upstream_time_millis,
+                                        &debug_name,
+                                    );
+                                    match res {
+                                        Ok(b) => b,
+                                        Err(err) => {
+                                            // XXX(chae): is it worth trying to use `tx_cap` if it exists or should just use `binary_cap`?
```

Comment:
I don't think we should be using the initial `binary_cap` anywhere in this operator because all the data that it is producing are a reaction on its inputs, which already come with capabilities. So we should just drop it on the floor during initialization (which happens automatically if you don't touch it).

I think our strategy should be that all data events are emitted with the corresponding `tx_cap` and all errors are emitted with the corresponding `data_cap`.